pkg/models/datasource_cache.go

```diff
@@ -201,22 +128,145 @@ func (ds *DataSource) GetHttpTransport() (*dataSourceTransport, error) {
 	return dsTransport, nil
 }
 
-func (ds *DataSource) sigV4Middleware(next http.RoundTripper) http.RoundTripper {
-	decrypted := ds.DecryptedValues()
+type proxyTransportCache2 struct {
+	cache map[int64]cachedRoundTripper2
+	sync.Mutex
+}
+
+type cachedRoundTripper2 struct {
+	updated      time.Time
+	roundTripper http.RoundTripper
+}
+
+var ptc2 = proxyTransportCache2{
```

Comment:
Is it wise to make this global, as opposed to a field on `DataSource` f.ex.? Asking because I don't know the context.

Reply:
Unfortunately yes because we want to have the cache outside of the actual `DataSource` struct since the struct holds one instance loaded from the database where's the cache could hold all instances loaded from the database that "haven't changed" - this allows us to cache HTTP clients for data sources until their configuration are changed. 

The goal is to convert core data sources to use SDK contracts and then those will not need to use this global cache or the DataSource struct. Until then I would like to continue use this legacy global cache or at least for getting this PR merged since it's already quite big.

Reply:
@marefr OK thanks for the clarification.